Source/WebCore/html/canvas/WebGLRenderingContextBase.cpp

```diff
@@ -5563,8 +5563,12 @@ void WebGLRenderingContextBase::maybeRestoreContext()
     if (!canvas)
         return;
 
-    if (!isContextLost())
+    if (!isContextLost()) {
         canvas->dispatchEvent(WebGLContextEvent::create(eventNames().webglcontextrestoredEvent, Event::CanBubble::No, Event::IsCancelable::Yes, emptyString()));
+        // Recreate the layer tree so that a new layerContentsDisplayDelegate() is picked up.
+        if (auto* renderer = canvas->renderer())
+            renderer->setNeedsLayout();
```

Comment:
It's unclear to me how this reconnects the "display delegate". Is it just forcing a layout to run and as a side-effect the display delegate is reconnected to the canvas?